### PR TITLE
Adding JobAgents API with SKU support to version 2023-05-01-preview

### PR DESCRIPTION
**Pull request by alejandro182**
# ARM (Control Plane) API Specification Update Pull Request 

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [ ] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [x] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I have reviewed the general guidance on the spec PR review process: https://aka.ms/specprreview.
- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data-plane related specifications.
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy).
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can request review from an ARM API Review board.

### ARM API changes review

- If you want for the ARM team to review this PR, you must add the `ARMReview` label. 
- The automation may automatically add the `ARMReview` label, if appropriate.  
  If this happens, proceed according to guidance given in GitHub comments also added by the automation.

### Breaking change review

If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), 
follow the process outlined in the [High-level Breaking Change Process doc](https://aka.ms/breakingchangesprocess#high-level-breaking-change-process).
      
## Getting help

- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For additional help, see https://aka.ms/azsdk/support/spectools.

### CODE REVIEW

**Review by jamestao — APPROVED** [review]
LGTM

**Review by jamestao — APPROVED** [review]
LGTM

**Comment by TimLovellSmith** [comment]
I notice the tickbox checked says '(This is applicable only when you are revising a private preview API version.)' but this looks like it might be an existing _public_ preview api version. Is that correct? @alejandro182

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json
Comment (TimLovellSmith) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:163]: >examples [](http://example.com/codeflow?start=14&length=8)

nice set of examples! <3

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json
Comment (TimLovellSmith) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:316]: >JobAgentIdentity [](http://example.com/codeflow?start=5&length=16)

It's best if we can reuse the common-types definitions for managed identity....

Comment (TimLovellSmith) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:316]: Please replace with references to appropriate definitions from `[https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/managedidentity.json`](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/managedidentity.json`) or reply to explain any problems with that.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json
Comment (TimLovellSmith) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:418]: >/v1/types.json#/definitions/Sku [](http://example.com/codeflow?start=34&length=31)

its also there in v5, so I'd just suggest using the general purpose v5 definition or... (can't believe I'm saying this but) declaring your own sku type. Because often RPs really can document their own SKU type much better than a generic documentation would.

[https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/types.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/types.json)

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json
Comment (TimLovellSmith) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:430]: >JobAgentUserAssignedIdentity [](http://example.com/codeflow?start=5&length=28)

Please replace with references to appropriate definitions from [[https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/managedidentity.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/managedidentity.json)](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/managedidentity.json`) or reply to explain any problems with that.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json
Comment (TimLovellSmith) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:385]: I'm not sure these lines about allowedResources empty add anything, do we need it?

**Comment by TimLovellSmith** [comment]
>     "ServerNameParameter": {

consider adding 'pattern' to document your existing server-side validations

---
Refers to: specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:450 in f793b05. [](commit_id = f793b058c2df2fbd1033e16ef850b0fd04e143b1, deletion_comment = False)

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json
Comment (TimLovellSmith) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:47]: >default [](http://example.com/codeflow?start=11&length=7)

you should also declare the schema of the default [error] response. It should be declared as a `$ref` to ErrorResponse here

[https://github.com/Azure/azure-rest-api-specs/blob/32c178f2467f792a322f56174be244135d2c907f/specification/common-types/resource-management/v5/types.json#L303](https://github.com/Azure/azure-rest-api-specs/blob/32c178f2467f792a322f56174be244135d2c907f/specification/common-types/resource-management/v5/types.json#L303)

**Comment by TimLovellSmith** [comment]
>           "$ref": "../../../common/v1/types.json#/definitions/TrackedResource"

Consider updating to v5 common types TrackedResource.

---
Refers to: specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:297 in f793b05. [](commit_id = f793b058c2df2fbd1033e16ef850b0fd04e143b1, deletion_comment = False)

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json
Comment (alejandro182) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:316]: Hi Tim, this was added by merging with new AAD changes, @rayboyd30 did managed identity presented any problems?

Comment (alejandro182) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:316]: this is the other PR changes
https://github.com/Azure/azure-rest-api-specs/pull/25553/files#diff-269ee4de58bde0815574778fe9ccc75fdb419beedef3e04f7f5d791f15fe4b2b

**Comment by alejandro182** [comment]
Hi @TimLovellSmith 
Seems like PR had previously being signed off but since the new branch release was created, it was modified to merge to it 
image

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json
Comment (TimLovellSmith) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:316]: Looks like that PR hasn't been reviewed yet? Are you trying to club two reviews together here? It might in fact be simpler to get it finished that way...

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json
Comment (alejandro182) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:47]: These seems to have been autogenerated by engine by list of ErrorDescriptions

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json
Comment (alejandro182) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:418]: This seems to have been autopopulated, and seems to be using on other places of the repo
https://github.com/Azure/azure-rest-api-specs/blob/release-sql-Microsoft.Sql-2023-05-01-preview/specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/Databases.json

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json
Comment (alejandro182) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:385]: This was also added by engine

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json
Comment (TimLovellSmith) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:47]: I don't know what engine you're using, but could you figure out a way to fix, or further configur eit?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json
Comment (alejandro182) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:316]: Seems like this would require extra effort to define common type for SQL. From other conversation:

[rayboyd30](https://github.com/rayboyd30) [5 days ago](https://github.com/Azure/azure-rest-api-specs/pull/25553/files#r1331943597)
aligning with SQL common type def needs extra efforts from SQL security team. We have to ask for an exception in current version. we will coordinate with SQL security team to fix it in next version. But for this version, please unblock.
Here is the work item to align common types: https://msdata.visualstudio.com/Database%20Systems/_workitems/edit/2697929

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json
Comment (alejandro182) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:430]: From other conversation: 
We would like to keep this consistent with other Microsoft.Sql APIs such as: https://github.com/Azure/azure-rest-api-specs/blob/def187e2e78d7173d8fdd7f77740dd9719e1dfbf/specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/Databases.json#L875C7-L875C7

File: specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json
Comment (raosuhas) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json:316]: please wirk with your team to move the identities schema into the SQL common types. It is not efficient to keep redifining it everywhere. Signing off on this for now.

**Comment by raosuhas** [comment]
please fix the tag : APIStewardshipBoard-SignedOff required

**Comment by rkmanda** [comment]
Please get a sign off for the data plane changes and then add the MergeRequested label

**Review by ericshape — APPROVED** [review]
this is control plane API change. we don't need data plane approval.

**Comment by sjanamma** [comment]
@alejandro182 - please make sure your PR has met the merge requirements -https://github.com/Azure/azure-rest-api-specs/pull/24995/checks?check_run_id=17190976176

**Comment by ericshape** [comment]
@sjanamma 
do we know how to get the approval from APIStewardshipBoard? do you know who we need to tag? 
this is a control plane/management service API change.  do we know why does it needs a data plane sign-off?

**Comment by jacobopac** [comment]
I added back the approved-sdkbreakingchange-python as I think Ji accidentally removed it. Now we just need the data plane sign-off.

**Comment by jacobopac** [comment]
Removing the data-plane tag per expert comment 

[2:19 PM] Suhas Rao
looks like itis incorrectly flagged 

[2:19 PM] Suhas Rao
you can just ask her to ignore that 

[2:19 PM] Suhas Rao
if it was data plane then changes would not be in that folder

**Comment by raosuhas** [comment]
> please fix the tag : APIStewardshipBoard-SignedOff required

INcorrectly flagged. Ignoring this

**Review by raosuhas — APPROVED** [review]

**Pull request merged**
